src/Workspaces/CoreTest/UtilityTest/IntervalTreeTests.cs

```diff
@@ -416,4 +416,31 @@ public void TestProperBalancing()
             AssertEx.Equal(tree, Enumerable.Range(1, i));
         }
     }
+
+    [Fact]
+    public void TestVeryLargeBalancing()
```

Comment:
i wnated tests that also validated the edge cases in large trees to ensure the math was scaling up properly.